gulpfile.js

```diff
@@ -0,0 +1,187 @@
+  , exec = require('child_process').exec
+
+var paths = {
+  css: './app/_scss', // no globbing
+  images: ['./app/img/**/*'], // doesnt really matter
+  svg: ['./app/img/svg/**/*'], // once there will be some
+  fonts: ['./app/fonts'], // will come later
+  js: ['./app/js/index.js'] // browserify
+}
+
+var production = false // set true if in production mode
+
+// development serve
+gulp.task('serve', ['jekyll', 'css', 'js'], function () {
+  browserSync.init({
+    server: {
+      baseDir: ['.jekyll', '.tmp', 'app'],
+    },
+    port: 4000
+  })
+
+  // watch and recompiles files
+  gulp.watch([
+    '_config.yml',
+    'app/**/*.{html,yml,md,mkd,markdown,json}',
+    '!app/_bower_components/**/*'
+  ], ['jekyll', reload])
+  gulp.watch([paths.css + '**/*.scss'], ['css'])
+  gulp.watch(['app/js/**/*.js'], ['jshint'])
+  // browserify watches the javascripts
+
+})
+
+
+// serve production stuff
+// DONT USE ON SERVER...I think
+gulp.task('serve:dist', ['production'], function () {
+  browserSync.init({
+    server: {
+      baseDir: 'dist'
+    },
+    port: 4000
+  })
+})
+
+// scss -> autoprefixer -> good ol' css
+gulp.task('css', function () {
+  var options = { bundleExec: true, require: ['bourbon', 'neat'] }
+  options.sourcemap = production ? false : true
+  return $.sass(paths.css, options)
+    .on('error', $.gutil.log.bind($.gutil))
+    .pipe($.prefix())
+    .pipe($.if(!production, $.sourcemaps.write()))
+    .pipe(gulp.dest('.tmp/css'))
+    .pipe($.if(!production, reload({ stream: true })))
+    .pipe($.if(!production, $.size({ title: 'styles' })))
+})
+
+// javascripts with browserify
+gulp.task('js', function () {
+  var bOptions = { entries: [paths.js], debug: true }
+  var b = production ? browserify(bOptions) :
+    watchify(browserify(bOptions), watchify.args)
+  if (!production) b.on('update', bundle)
+  function bundle () {
+    return b.bundle()
+      .pipe(source('script.js'))
+      .pipe(buffer())
+      .pipe($.sourcemaps.init({ loadMaps: true }))
+      .pipe($.if(!production, $.sourcemaps.write()))
+      .pipe(gulp.dest('.tmp/js'))
+      .pipe($.if(!production, reload({ stream: true })))
+      .pipe($.if(!production, $.size({ title: 'javascripts' })))
+  }
+  return bundle()
+})
+
+// jekyll
+gulp.task('jekyll', function (done) {
+  if (production) {
+    exec('bundle exec jekyll build --source app --config _config.yml,_config.build.yml --destination dist', done)
+  } else {
+    exec('bundle exec jekyll build --source app --config _config.yml --destination .jekyll', done)
+  }
+})
+
+// Compiles HTML
+// Takes a LONG time
+gulp.task('html', function () {
+
+  var jsFilter = $.filter('**/*.js')
+    , cssFilter = $.filter('**/*.css')
+    , htmlFilter = $.filter('**/*.html')
+    , indexFilter = $.filter('index.html')
+
+  var assets = $.useref.assets({ searchPath: '{.tmp,app}' })
+
+  return gulp.src('dist/**/*.html')
+    .pipe(indexFilter) // only source from index.html
```

Comment:
Production speeds now increased drastically due to sourcing of only index.html when looking for assets.